test/instances/morpho-blue/MorphoBlue_wstETH_USDC_Hyperdrive.t.sol

```diff
@@ -0,0 +1,818 @@
+import { LPMath } from "../../../contracts/src/libraries/LPMath.sol";
+import { ERC20ForwarderFactory } from "../../../contracts/src/token/ERC20ForwarderFactory.sol";
+import { ERC20Mintable } from "../../../contracts/test/ERC20Mintable.sol";
+import { InstanceTest } from "../../utils/InstanceTest.sol";
+import { HyperdriveUtils } from "../../utils/HyperdriveUtils.sol";
+import { Lib } from "../../utils/Lib.sol";
+
+contract WSTETH_USDC_MorphoBlueHyperdriveTest is InstanceTest {
+    using FixedPointMath for uint256;
+    using HyperdriveUtils for IHyperdrive;
+    using MarketParamsLib for MarketParams;
+    using MorphoBalancesLib for IMorpho;
+    using Lib for *;
+    using stdStorage for StdStorage;
+
+    // The mainnet Morpho Blue pool.
+    IMorpho internal constant MORPHO =
+        IMorpho(0xBBBBBbbBBb9cC5e90e3b3Af64bdAF62C37EEFFCb);
+
+    // The ID of the wstETH/USDC market.
+    bytes32 internal constant MARKET_ID =
+        bytes32(
+            0xb323495f7e4148be5643a4ea4a8221eef163e4bccfdedc2a6f4696baacbc86cc
+        );
+
+    // The address of the loan token. This is just the DAI token.
+    address internal constant LOAN_TOKEN =
+        address(0xA0b86991c6218b36c1d19D4a2e9Eb0cE3606eB48);
+
+    // The address of the collateral token. This is just the wstETH token.
+    address internal constant COLLATERAL_TOKEN =
+        address(0x7f39C581F595B53c5cb19bD0b3f8dA6c935E2Ca0);
+
+    // The address of the oracle.
+    address internal constant ORACLE =
+        address(0x48F7E36EB6B826B2dF4B2E630B62Cd25e89E40e2);
+
+    // The address of the interest rate model.
+    address internal constant IRM =
+        address(0x870aC11D48B15DB9a138Cf899d20F13F79Ba00BC);
+
+    // The liquidation loan to value ratio of the wstETH/USDC market.
+    uint256 internal constant LLTV = 860000000000000000;
+
+    // Whale accounts.
+    address internal LOAN_TOKEN_WHALE =
+        address(0x4B16c5dE96EB2117bBE5fd171E4d203624B014aa);
+    address[] internal baseTokenWhaleAccounts = [LOAN_TOKEN_WHALE];
+
+    // The configuration for the instance testing suite.
+    InstanceTestConfig internal __testConfig =
+        InstanceTestConfig({
+            name: "Hyperdrive",
+            kind: "MorphoBlueHyperdrive",
+            decimals: 6,
+            baseTokenWhaleAccounts: baseTokenWhaleAccounts,
+            vaultSharesTokenWhaleAccounts: new address[](0),
+            baseToken: IERC20(LOAN_TOKEN),
+            vaultSharesToken: IERC20(address(0)),
+            // NOTE: The share tolerance is quite high for this integration
+            // because the vault share price is ~1e12, which means that just
+            // multiplying or dividing by the vault is an imprecise way of
+            // converting between base and vault shares. We included more
+            // assertions than normal to the round trip tests to verify that
+            // the calculations satisfy our expectations of accuracy.
+            shareTolerance: 1e8,
```

Comment:
seems like this is lower than the other morpho test share tolerances

Reply:
You mean higher relative to the token decimals? Yes it seems to be a precision issue, but I'll take another look to see if I can bring it down.

Reply:
I was able to drop this to 1e3 which seems acceptable.